doc/recognition.xml

```diff
@@ -436,6 +445,19 @@ elements that are generated from them remember, how they were acquired.
 </Description>
 </ManSection>
 
+<ManSection>
+<Attr Name="ValidateHomomInput" Arg="ri"/>
+<Description>
+    The value of this attribute is a function. This function returns
+    a boolean which is <K>false</K> when passed a value which is not
```

Comment:
Is this really what we want? This suggests that `ValidateHomomInput` is completely testing membership of the element in the source.

I had assumed that we want a looser check. My assumption is that `ValidateHomomInput` would be required return `false` if either:
1. it is not in the source and this fact is easy to compute;
2. it is not in the source, and applying the homomorphism will produce an error/incorrect result.

and return `true` otherwise (of course, it could do more work if it wanted).

This would still allow the homomorphism to be given values for which it will correctly return `fail`.
What do you think about this?

Reply:
You are right, that's far too strong.

In general, all we really care about is the Homom works. In practice there is usually (as described elsewhere) a property which this node of the tree uses to split the group (invariant subspace, block structure, ...), and we want to check that property.

Reply:
To add to that: at the beginning of the recognition we can't expect to have a membership test for the group since that's precisely what we are trying to construct.

Reply:
Wait, I've thought about this some more, I think it is right (although I can't remember if I've changed it & pushed since this comment was first made)

Note that Homom usually accepts many things which are not in the group. Things not in the image of Homom should not be given to Homom as an argument, so we have to reject them.

Reply:
I see your point, assuming that you mean 'source' rather than 'image'. I just wanted to be sure that we're not insisting on a group membership test.